### PR TITLE
Fixup for OS selection in GitHub Actions unittests workflow

### LINKED ISSUE #627: GitHub Actions: unit tests not exercising different operating systems
For example, [this `py39` `windows-latest` build job](https://github.com/hhursev/recipe-scrapers/actions/runs/3215401157/jobs/5256472876) seems to be using an [unexpected version of Windows](https://github.com/hhursev/recipe-scrapers/actions/runs/3215401157/jobs/5256472876#step:1:3).

### PR DESCRIPTION
**Pull request by jayaddison**
Resolves #627

### CODE REVIEW

File: .github/workflows/unittests.yaml [hunk 1]
```diff
@@ -10,11 +10,11 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-latest
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
         python-version: ["3.7", "3.8", "3.9", "3.10", "3.11.0-rc.1"]
```
Comment (jayaddison) [inline .github/workflows/unittests.yaml:16]: I think that the {`windows-latest`, `3.11.0-rc.1`} build [install-time failure for `lxml`](https://github.com/hhursev/recipe-scrapers/actions/runs/3221015714/jobs/5268442428#step:4:262) is due to the fact that no precompiled wheels are available for that {package, Python version, OS} combination on PyPi.

`macos-latest` did appear to manage to [install `lxml` successfully](https://github.com/hhursev/recipe-scrapers/actions/runs/3221015714/jobs/5268442100#step:4:172) under Py3.11-rc.1 (before the job was cancelled due to the Windows check failure)

Probably easiest to add an exclusion condition for the Windows + Py3.11-rc1 combination to resolve this.

File: .github/workflows/unittests.yaml [hunk 2]
```diff
@@ -10,11 +10,16 @@ on:
...
+        exclude:
+          - os: macos-latest
+            python-version: "3.11.0-rc.1"
```
Comment (jayaddison) [inline .github/workflows/unittests.yaml:19]: Strangely, all of the `macos-latest` builds succeeded except for one running this release candidate version of Py3.11.

For that version, schema data extraction [failed](https://github.com/hhursev/recipe-scrapers/actions/runs/3221154009/jobs/5268750127#step:5:26) for the generic `wild_mode` scraper test.

Comment (jayaddison) [reply .github/workflows/unittests.yaml:19]: I'm having trouble figuring this one out.  Despite enabling debug logging, there wasn't much additional context in the `wild_mode` scraper test failure output.

Comment (jayaddison) [reply .github/workflows/unittests.yaml:19]: Roughly speaking: it's as if the unit test doesn't read the HTML content correctly (or at all?) when the [`TestWildMode`](https://github.com/hhursev/recipe-scrapers/blob/93c0c86a4a27697d34efcc1d04d9531af2df6223/tests/test_wild_mode.py) test runs.

The errors arise when the scraper-under-test attempts to retrieve parsed schema.org JSON-LD fields from the content.  For example, we [see](https://github.com/hhursev/recipe-scrapers/actions/runs/3221492074/jobs/5269493368#step:5:5790):

```
  File "/Users/runner/work/recipe-scrapers/recipe-scrapers/recipe_scrapers/_factory.py", line 21, in cook_time
    return self.schema.cook_time()
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/work/recipe-scrapers/recipe-scrapers/recipe_scrapers/_schemaorg.py", line 120, in cook_time
    raise SchemaOrgException("Cooktime information not found in SchemaOrg")
recipe_scrapers._exceptions.SchemaOrgException: recipe-scrapers exception: Cooktime information not found in SchemaOrg
```

Note that this is happening in Py3.10 as well as seen in the [latest builds](https://github.com/hhursev/recipe-scrapers/actions/runs/3221492074/jobs/5269493368#step:5:22).

**Comment by jayaddison** [comment]
Merging the changes here so that subsequent pull requests will run with the full intended list of operating systems.  Investigation into the test failures for Py3.10 / Py3.11 on MacOS continues in #629.

File: .github/workflows/unittests.yaml [hunk 3]
```diff
@@ -10,11 +10,18 @@ on:
 
 jobs:
   test:
-    runs-on: ubuntu-latest
```
Comment (hhursev) [inline .github/workflows/unittests.yaml:13]: that's been pretty lame on my part 😅

**Pull request merged**